### DISCUSSION TITLE
Changing Seed?

### DISCUSSION THREAD

**Discussion by n1bor** [ACINQ/eclair#2548] [Node operators]
Am I right in thinking there is no way to change the node or channel seed without closing all the channels?

And if there is even a tiny chance that these seeds have leaked one should change them? As with them someone can steel all the funds on the channels by paying it to themselves?

Has there been any discussions proposals on how it could be made possible to change them? Just having any key that can not be changed is always going to present problems at some point for operators.

**Comment by t-bast** [comment]
There's indeed no way to migrate channels to a new seed without on-chain transactions. If your seed has been compromised, you need to close all channels and start a new node.

> Has there been any discussions proposals on how it could be made possible to change them? Just having any key that can not be changed is always going to present problems at some point for operators.

I don't believe this is possible, there's a fundamental issue here that past transactions can always be broadcast, so even if you migrated channels to use different keys for future transactions, an attacker that gets the old keys would still be able to broadcast revoked transactions.
